raptor/raptor.py

```diff
@@ -44,26 +47,60 @@ def __init__(self, app, binary):
         except NotImplementedError:
             self.profile = None
 
+        # Create the runner
+        cmdargs = []
+        if app == 'chrome':
+            cmdargs.append('--load-extension={}'.format(
+                           os.path.join(webext_dir, 'raptor-chrome')))
+
+        self.output_handler = OutputHandler()
+        process_args = {
+            'processOutputLine': [self.output_handler],
+        }
+        runner_cls = runners[app]
+        self.runner = runner_cls(
+            binary, cmdargs=cmdargs, profile=self.profile, process_args=process_args)
+
     def start_control_server(self):
         self.control_server = RaptorControlServer()
         self.control_server.start()
 
-    def run_test(self, test):
+    def run_test(self, test, timeout=None):
         gen_test_url(self.app, test)
         install_webext(self.app, self.profile)
-        start_browser(self.app, self.binary, self.profile)
+        self.runner.start()
+        first_time = int(time.time()) * 1000
+        proc = self.runner.process_handler
+        self.output_handler.proc = proc
 
-    def process_results(self):
+        try:
+            self.runner.wait(timeout)
+        finally:
+            try:
+                self.runner.check_for_crashes()
+            except NotImplementedError:  # not implemented for Chrome
+                pass
+
+        if self.runner.is_running():
+            self.log("Application timed out after {} seconds".format(timeout))
+            self.runner.stop()
+
+        proc.output.append(
+            "__startBeforeLaunchTimestamp%d__endBeforeLaunchTimestamp"
+            % first_time)
+        proc.output.append(
+            "__startAfterTerminationTimestamp%d__endAfterTerminationTimestamp"
+            % (int(time.time()) * 1000))
+
+        self._process_results(proc.output)
+
+    def _process_results(self, output):
```

Comment:
yeah so the results won't come from the browser pid output handler; later on I will be adding a method to the control server that will be called here to grab the results as received by the control server